asm_to_pil/src/vm_to_constrained.rs

```diff
@@ -585,10 +585,8 @@ impl<T: FieldElement> ASMPILConverter<T> {
             Expression::FunctionCall(_) => panic!(),
             Expression::Reference(reference) => {
                 // TODO check it actually is a register
-                vec![(
-                    1.into(),
-                    AffineExpressionComponent::Register(reference.name),
-                )]
+                let name = reference.try_to_identifier().unwrap();
+                vec![(1.into(), AffineExpressionComponent::Register(name.clone()))]
```

Comment:
I know it's unrelated to this PR, but do you know what this hardcoded 1 is?

Reply:
Yes, it's the factor one. The return type is to be interpreted roughly as `v.map(|(f, x)| f * x).sum()`. Maybe we could create a custom data structure with operators at some point.